nvflare/app_common/aggregators/intime_accumulate_model_aggregator.py

```diff
@@ -12,180 +12,118 @@
+            self.expected_data_kind = expected_data_kind
+        else:
+            if expected_data_kind not in [DataKind.WEIGHT_DIFF, DataKind.WEIGHTS]:
+                raise ValueError(f"expected_data_kind={expected_data_kind} not in WEIGHT_DIFF or WEIGHTS")
+            self.expected_data_kind = {self._single_dxo_key: expected_data_kind}
+
+        # Set up DXO aggregators
+        self.dxo_aggregators = dict()
+        for key in self.expected_data_kind.keys():
+            self.dxo_aggregators.update({key: DXOAggregator(exclude_vars=exclude_vars)})
 
     def accept(self, shareable: Shareable, fl_ctx: FLContext) -> bool:
         """Store shareable and update aggregator's internal state
 
         Args:
-            shareable: information from client
+            shareable: information from contributor
             fl_ctx: context provided by workflow
 
         Returns:
             The first boolean indicates if this shareable is accepted.
-            The second bollean indicates if aggregate can be called.
+            The second boolean indicates if aggregate can be called.
         """
         try:
             dxo = from_shareable(shareable)
-        except:
+        except BaseException:
             self.log_exception(fl_ctx, "shareable data is not a valid DXO")
             return False
         assert isinstance(dxo, DXO)
 
-        if dxo.data_kind not in (DataKind.WEIGHT_DIFF, DataKind.WEIGHTS):
-            self.log_error(fl_ctx, "cannot handle data kind {}".format(dxo.data_kind))
-            return False
-
-        if dxo.data_kind != self.expected_data_kind:
-            self.log_error(fl_ctx, "expect {} but got {}".format(self.expected_data_kind, dxo.data_kind))
-            return False
-
-        processed_algorithm = dxo.get_meta_prop(MetaKey.PROCESSED_ALGORITHM)
-        if processed_algorithm is not None:
-            self.log_error(fl_ctx, f"unable to accept shareable processed by {processed_algorithm}")
+        if dxo.data_kind not in (DataKind.WEIGHT_DIFF, DataKind.WEIGHTS, DataKind.COLLECTION):
+            self.log_error(
+                fl_ctx,
+                f"cannot handle data kind {dxo.data_kind}, "
+                f"expecting DataKind.WEIGHT_DIFF, DataKind.WEIGHTS, or DataKind.COLLECTION.",
+            )
             return False
 
-        current_round = fl_ctx.get_prop(AppConstants.CURRENT_ROUND)
-        self.log_debug(fl_ctx, f"current_round: {current_round}")
-        client_name = shareable.get_peer_prop(key=ReservedKey.IDENTITY_NAME, default="?")
-        contribution_round = shareable.get_header(AppConstants.CONTRIBUTION_ROUND)
+        # Accept expected DXO(s) in shareable
+        for key in self.expected_data_kind.keys():
+            if key == self._single_dxo_key:  # not expecting a single DXO
+                sub_dxo = dxo
+            else:  # expecting a collection of DXOs
+                sub_dxo = dxo.data.get(key)
```

Comment:
Need to check to make sure sub_dxo is actually a DXO here.

Reply:
Sure. I added a check.